main-serve-singleton.py

```diff
@@ -8,23 +8,21 @@
 import ray
 from ray import serve
 from starlette.requests import Request
-from whylogs.core.datasetprofile import DatasetProfile
+from whylogs.core import DatasetProfile, DatasetProfileView
 
 ray.init()
 serve.start()
 
-batch_size = 1000
-
 
 @ray.remote
-def log_frame(df: pd.DataFrame) -> DatasetProfile:
-    profile = DatasetProfile("")
-    profile.track_dataframe(df)
+def log_frame(df: pd.DataFrame) -> DatasetProfileView:
```

Comment:
return type here is inconsistent with what you return: DatasetProfile vs DatasetProfileView